cmake/packaging.cmake

```diff
@@ -1,36 +1,58 @@
 set(CPACK_PACKAGE_NAME "${PROJECT_NAME}")
-set(CPACK_PACKAGE_CONTACT "Luis Díaz Más <piponazo@gmail.com>")
+set(CPACK_PACKAGE_CONTACT "Luis DÃ­az MÃ¡s <piponazo@gmail.com>")
 set(CPACK_PACKAGE_VERSION ${PROJECT_VERSION})
 
-set(CPACK_SOURCE_GENERATOR TGZ)
 # https://libwebsockets.org/git/libwebsockets/commit/minimal-examples?id=3e25edf1ee7ea8127e941fd7b664e0e962cfeb85
 set(CPACK_SOURCE_IGNORE_FILES $(CPACK_SOURCE_IGNORE_FILES) "/.git/" "/build/" "\\\\.tgz$" "\\\\.tar\\\\.gz$" "\\\\.zip$" "/test/tmp/" )
 
+if ( MSVC )
+    set(CPACK_GENERATOR ZIP)  # use .zip - less likely to damage bin/exiv2.dll permissions
+else()
+    set(CPACK_GENERATOR TGZ)  # MinGW/Cygwin/Linux/MacOS-X etc use .tar.gz
+endif()
+
+set (BS "") # Bit Size
+if ( NOT APPLE )
+  if ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+    set (BS 64)
+  else()
+    set (BS 32)
+  endif()
+endif()
+
+set (LT "") # Library Type
+if ( NOT BUILD_SHARED_LIBS )
+	set (LT Static)
+endif()
+
+set (BT "") # Build Type
+if ( NOT ${CMAKE_BUILD_TYPE} STREQUAL Release )
+	set (BT ${CMAKE_BUILD_TYPE})
+endif()
+
 if ( MINGW OR MSYS )
-    if ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-        set (PACKNAME MinGW64)
-    else()
-        set (PACKNAME MinGW32)
-    endif()
     set (PACKDIR MinGW)
 elseif ( MSVC )
-    set (PACKNAME MSVC)
-    set (PACKDIR  MSVC)
-elseif ( APPLE OR LINUX OR CYGWIN )
-    set (PACKNAME ${CMAKE_SYSTEM_NAME}) # Darwin or Linux or CYGWIN
-    set (PACKDIR  ${PACKNAME})
+    set (PACKDIR msvc)
+elseif ( CYGWIN )
+    set (PACKDIR CYGWIN)
+elseif ( LINUX )
+    set (PACKDIR Linux)
+elseif ( APPLE )
+    set (PACKDIR Darwin)
 else()
-    set (PACKNAME Linux      ) # unsupported systems such as FreeBSD
-    set (PACKDIR  ${PACKNAME})
+    set (PACKDIR  Linux) # unsupported systems such as FreeBSD
 endif()
 
-if ( MSVC )
-    set(CPACK_GENERATOR ZIP)  # use .zip - less likely to damage bin/exiv2.dll permissions
-else()
-    set(CPACK_GENERATOR TGZ)  # MinGW/Cygwin/Linux/MacOS-X etc use .tar.gz
+set (CC "") # Compiler
```

Comment:
Few comments here. 
1) The default value of CC is empty. Is that the intended behaviour? In case the compiler is not clang, the value will remain empty.
2) The way to check for clang can be simplified (without creating variables for it):

```
    if (${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
        set (CC Clang)
    endif()
```

Reply:
Yes.  That's what I want.  We'll end up with bundle names such as:

_CYGWIN64ClangStaticDebug_

However, if for a SharedRelease built with GCC, the name will simply be:

_CYGWIN64_

I think there are 60 common build possibilities.
{Linux|CYGWIN|MinGW}{64|32}{''|Clang}{''|Static}{''|Debug} = 3x2x2x2x2 = 48
{Darwin}{''|Static}{''|Debug} = 2x2 = 4
{msvc}{64|32}{''|Static}{''|Debug} = 8
Total = 60

If I decide to to build with the other versions of Visual Studio {2008|2010|2012|2013|2015), that will be 5x8=40 more builds. 

If we have a **FULL** build (webready, video) that would double it to 200.

There's no problem presenting and storing this on the buildserver.  http://exiv2.dyndns.org:8080/userContent/builds/Platform/

On the downloads page, we'll only offer 6 http://exiv2.dyndns.org/download.html
_Source_
_msvc64_
_Linux64_
_Darwin_
_MinGW64_
_CYGWIN64_

For RC1 and RC2, I published MinGW64 and MinGW32 on _'downloads'_ because we were experiencing different issues on those platforms.  However, we've put that behind us now, so I'll only offer MinGW64 with RC3.